pkg/extender/extender.go

```diff
@@ -827,3 +912,176 @@ func (e *Extender) processCSIExtPodPrioritizeRequest(
 		storklog.PodLog(pod).Errorf("Failed to encode response: %v", err)
 	}
 }
+
+func (e *Extender) processVirtLauncherPodPrioritizeRequest(
+	encoder *json.Encoder,
+	args schedulerapi.ExtenderArgs) error {
+	pod := args.Pod
+
+	liveVMMigrationInProgress, podBeingLiveMigrated, err := e.getLiveMigrationInfo(pod)
+	if err != nil {
+		return fmt.Errorf("unable to find liveMigrationInfo: %v", err)
+	}
+	vmiName := e.getVMIName(pod)
+	if vmiName == "" {
+		return fmt.Errorf("unable to find vmiName for pod: %v", pod.Name)
+	}
+
+	if liveVMMigrationInProgress {
+		storklog.PodLog(pod).Debugf("Live VM migration in progress: (vmiName: %v, podBeingLiveMigrated: %v)", vmiName, podBeingLiveMigrated.Name)
+	} else {
+		storklog.PodLog(pod).Debugf("No ongoing live VM migration in progress. vmiName: %v", vmiName)
+	}
+
+	vmi, err := kvd.Instance().GetVirtualMachineInstance(context.TODO(), pod.Namespace, vmiName)
+	if err != nil {
+		return fmt.Errorf("unable to find VMI Info: %v", err)
+	}
+	storklog.PodLog(pod).Debugf("VMI is using PVC %v: ", vmi.RootDiskPVC)
+
+	pvId, err := e.Driver.GetVolumeIDFromPVC(vmi.RootDiskPVC, pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("unable to inspect PVC: %v err: %v", vmi.RootDiskPVC, err)
+	}
+	storklog.PodLog(pod).Debugf("Volume id for scoring: %v", pvId)
+
+	volInfo, err := e.Driver.InspectVolume(pvId)
+	if err != nil {
+		return fmt.Errorf("unable to inspect volume id: %v. err: %v", pvId, err)
+	}
+
+	if liveVMMigrationInProgress {
+		podBeingLiveMigratedIsBindMounted, err := e.isUsingBindMount(podBeingLiveMigrated, volInfo)
+		if err != nil {
+			// Live migration in progress default is antihyperconvergence
+			storklog.PodLog(pod).Debugf("Unable to make determination if other pod is using bind mount. Using Antihyperconvergence")
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+			return nil
+		}
+		storklog.PodLog(pod).Debugf("Pod %v is running with same VMI owner reference and using localAttachment: %v", podBeingLiveMigrated.Name, podBeingLiveMigratedIsBindMounted)
+
+		if podBeingLiveMigratedIsBindMounted {
+			// Antihyperconvergence
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		} else {
+			// This means existing pod is using NFS mount.
+			// Use Antihyperconvergence and give high score to volume with
+			// attached node as an exception
```

Comment:
As explained in a statement above, anti hyperconvergence is chosing non replica nodes. But here we also have  
preferLocalAttachment = true will end up giving 2x score to local nodes. x score to non replica nodes. And default score to other replica nodes(with volume not attached) to avoid multiple migrations when volume attachments move.